src/backend/env/memory.rs

```diff
@@ -304,29 +304,42 @@ impl Allocator {
 }
 
 #[derive(Default, Serialize, Deserialize)]
-pub struct ObjectManager<K: Ord + Eq> {
+pub struct ObjectManager<K: Ord + Eq, T: Serialize + DeserializeOwned> {
     index: BTreeMap<K, (u64, u64)>,
     #[serde(skip)]
     api: Rc<RefCell<Api>>,
+    #[serde(skip)]
+    phantom: std::marker::PhantomData<T>,
 }
 
-impl<K: Eq + Ord + Clone + Display> ObjectManager<K> {
+impl<K: Eq + Ord + Clone + Copy + Display, T: Serialize + DeserializeOwned> ObjectManager<K, T> {
     pub fn len(&self) -> usize {
         self.index.len()
     }
 
-    pub fn insert<T: Serialize>(&mut self, id: K, value: T) -> Result<(), String> {
+    pub fn insert(&mut self, id: K, value: T) -> Result<(), String> {
         self.index.insert(id, self.api.borrow_mut().write(&value)?);
         Ok(())
     }
 
-    pub fn get<T: DeserializeOwned>(&self, id: &K) -> Option<T> {
+    pub fn get(&self, id: &K) -> Option<T> {
         self.index
             .get(id)
             .map(|(offset, len)| self.api.borrow().read(*offset, *len))
     }
 
-    pub fn remove<T: DeserializeOwned>(&mut self, id: &K) -> Result<T, String> {
+    pub fn iter(&self) -> Box<dyn DoubleEndedIterator<Item = (K, T)> + '_> {
```

Comment:
Not yet. I played locally with moving users and the ledger to the stable memory and it would definitely be needed there, so I thought I can add it already because it's the second time I am adding it and always spend like 10 minutes until it works... 😅 So we will need it very soon anyway. But I am also fine removing it if you have any concerns.